zerver/lib/actions.py

```diff
@@ -2980,9 +2980,34 @@ def do_remove_streams_from_default_stream_group(realm: Realm, group_name: Text,
     group.save()
     notify_default_stream_groups(realm)
 
+def do_change_default_stream_group_name(realm: Realm, group_name: Text, new_group_name: Text) -> None:
+    if group_name == new_group_name:
+        raise JsonableError(_("'%s' is already the default stream group name. Please provide another name")
+                            % (group_name,))
+
+    if not DefaultStreamGroup.objects.filter(name=new_group_name, realm=realm).exists():
```

Comment:
It'd be better to raise the exception if the new group name already exists first; you save indenting all the code below for now reason.